xonsh/prompt/vc_branch.py

```diff
@@ -9,49 +9,40 @@
 import subprocess
 
 import xonsh.platform as xp
+import xonsh.prompt
+
+import threading, queue
+
+
+def _get_git_branch(q):
+    try:
+        status = subprocess.check_output(['git', 'status'],
+                                         stderr=subprocess.DEVNULL)
+    except subprocess.CalledProcessError:
```

Comment:
I think that if you caught OSError here too, you could avoid the error later on. You might also need to use PIPE, but let'd try one thing at a time: http://stackoverflow.com/questions/18439712/subprocesss-popen-closes-stdout-stderr-filedescriptors-used-in-another-thread-w